### PR TITLE
Change definition of eltype and add tests

### LINKED ISSUE #263: Change eltype(x::T) to return T
The current definition of `eltype` is causing broadcasting to fail:
```
julia> using IntervalArithmetic

julia> x = 3..4
[3, 4]

julia> x .* [1]
ERROR: MethodError: no method matching Float64(::Interval{Float64})
```
For this to work, we need `eltype(x::T) where {T<:Interval} = T`, and similarly for `IntervalBox`.

As far as I can tell we don't use `eltype` anywhere, so this should be fine.

### PR DESCRIPTION
**Pull request by krish8484**
Closes #263

### CODE REVIEW

**Comment by dpsanders** [comment]
Any particular reason for making a new file elt.jl? If you do make a new file it should have a name that is more understandable.

Also, I suggest making a new branch in your fork to make pull requests from.

**Comment by krish8484** [comment]
Functions at `intervals.jl` were interfering with the `eltype` function, so I created a new file for eltype functions namely `elt.jl`. Name will be changed if it is not okay.
Do you have an idea of what is causing the build test to fail?

**Comment by krish8484** [comment]
Works only for Intervals now, let me know if the name of the file needs to be changed.

File: src/intervals/elt.jl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+Base.eltype(x::Interval{T}) where {T<:Real} = T
+
+# Base.eltype(x::T) where {T<:IntervalBox} = T
```
Comment (dpsanders) [inline src/intervals/elt.jl]: We need a definition for IntervalBox as well.

Comment (dpsanders) [reply src/intervals/elt.jl]: I really don't see the need for a separate file this short.

File: test/runtests.jl [hunk 2]
```diff
@@ -1,4 +1,3 @@
-
 using IntervalArithmetic
 if VERSION < v"0.7.0-DEV.2004"
```
Comment (dpsanders) [inline test/runtests.jl]: We can remove these old VERSION checks now since we require some version > 1.

Comment (dpsanders) [reply test/runtests.jl]: Maybe that could/should be a separate PR though.

**Comment by dpsanders** [comment]
LGTM, thanks!

You will need to bump the patch version number too.

**Comment by dpsanders** [comment]
Thanks!

**Pull request merged**